tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_bitwise_not.h

```diff
@@ -0,0 +1,43 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ckernel.h"
+#include "ckernel_defs.h"
+#include "sfpi.h"
+#include "noc_nonblocking_api.h"
+
+using namespace sfpi;
+
+namespace ckernel {
+namespace sfpu {
+
+template <bool APPROXIMATION_MODE, int ITERATIONS = 8>
+inline void calculate_bitwise_not(const uint value) {
+#pragma GCC unroll 0
+    for (int d = 0; d < ITERATIONS; d++) {
+        vInt input = dst_reg[0];
+        vInt val = input;
+        vInt res;
+
```

Comment:
There is an inverse operation in sfpi. It should be this: `~`.
Can you please try doing this:

```
vInt input = dst_reg[0];
vInt result = ~input;
```
and let me know if that does or doesnt work?

Reply:
The inverse operation `~` does not give correct outputs for negative inputs.
```
    vInt input = dst_reg[0];
    vInt res = ~input;
    res = 0 - res;
```
Eg: For input -1, the actual output is 0, but the above snippet returns -2 as the output.
I used vFloat in order to handle the negative input range instead of the inverse operation `~`.

Reply:
Ok I see the issue here, pytorch operates on 2's complement integers, while the SFPU uses sign + magnitude integers. For this operation, I think it can be simplified. For example I think for the negative values. Doing:
```
 v_if(input < 0) {
      vInt unsigned_mag = input & ~(1 << 32); //This will mask off the signed bit for 32 bit integers, you can also do this with setsgn operation
      res = unsigned_mag - 1;  //This does a reverse of a 2s complement conversion 
 }
```
This way you can keep the precision for int32, and also its a faster operation that doing 2 conversions. Please let me know if that works